### PR TITLE
ENG-159: Journal cash receipts from collection confirmations

### PR DESCRIPTION
**Pull request by Connorbelez**
# Wire cash ledger posting pipeline into collection confirmation flow

Wire the cash ledger posting pipeline into the collection confirmation
flow so each obligation payment and any overpayment surplus produce
double-entry cash ledger entries. Adds postingGroupId to correlate
entries within a single collection attempt, routes overpayments to
UNAPPLIED_CASH, and gracefully skips posting when no receivable
account exists (deferring to ENG-156 reconciliation).

## Changes

- **Overpayment routing**: Routes excess payment amounts beyond obligation balances to UNAPPLIED_CASH via new `postOverpaymentToUnappliedCash` function
- **PostingGroupId correlation**: Generates `cash-receipt:${attemptId}` postingGroupId to correlate all journal entries within a single collection attempt
- **Graceful receivable handling**: Replaces `requireCashAccount` with `findCashAccount` for BORROWER_RECEIVABLE lookups, logging warnings and skipping posting when accounts don't exist
- **CASH_RECEIVED family expansion**: Adds UNAPPLIED_CASH to credit families for CASH_RECEIVED entry type to support overpayment entries
- **Comprehensive test coverage**: Adds unit tests for cash receipt edge cases and integration tests for the full confirmation flow including multi-obligation payments and overpayment scenarios

Co-Authored-By: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

## Release Notes

* **New Features**
  * Overpayments from collection confirmations are now automatically routed to unapplied cash accounts
  * Payment receipts are tracked with posting group identifiers for improved correlation and auditing

* **Bug Fixes**
  * Enhanced payment processing to gracefully handle missing borrower receivable accounts with improved logging

* **Tests**
  * Added comprehensive unit and integration tests for cash receipt processing and overpayment handling

### CODE REVIEW

File: convex/payments/cashLedger/__tests__/cashReceipt.test.ts [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+	mortgageId: Id<"mortgages">,
+	initialDebit: bigint
+) {
+	return t.run(async (ctx) => {
+		const account = await getOrCreateCashAccount(ctx, {
+			family: "UNAPPLIED_CASH",
+			mortgageId,
+		});
+		await ctx.db.patch(account._id, { cumulativeDebits: initialDebit });
+		return account._id;
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/cashLedger/__tests__/cashReceipt.test.ts]: Fixed in a6a95a1. The `seedUnappliedCashAccount` helper has been removed from both `cashReceipt.test.ts` and `cashReceiptIntegration.test.ts`. Since UNAPPLIED_CASH is now debited (not credited) in the overpayment posting, its debit-normal balance naturally increases from zero without tripping the guard. No more direct mutation of internal balance fields needed.
